cli/azd/pkg/ext/hooks_manager.go

```diff
@@ -114,3 +123,68 @@ func (h *HooksManager) filterConfigs(
 
 	return matchingHooks, nil
 }
+
+// HookValidationResult contains warnings found during hook validation
+type HookValidationResult struct {
+	Warnings []HookWarning
+}
+
+// HookWarning represents a validation warning for hooks
+type HookWarning struct {
+	Message string
+}
+
+// ValidateHooks validates hook configurations and returns any warnings
+func (h *HooksManager) ValidateHooks(ctx context.Context, allHooks map[string][]*HookConfig) *HookValidationResult {
+	result := &HookValidationResult{
+		Warnings: []HookWarning{},
+	}
+
+	hasPowerShellHooks := false
+
+	// Check all hooks
+	for _, hookConfigs := range allHooks {
+		for _, hookConfig := range hookConfigs {
+			if hookConfig.IsPowerShellHook() {
+				hasPowerShellHooks = true
+				break
+			}
+		}
+		if hasPowerShellHooks {
+			break
+		}
+	}
+
+	// If we found PowerShell hooks, check if pwsh is available
+	if hasPowerShellHooks {
+		if err := h.commandRunner.ToolInPath("pwsh"); errors.Is(err, osexec.ErrNotFound) {
+			var warningMessage string
+
+			// Check if legacy powershell is available
+			if powershellErr := h.commandRunner.ToolInPath("powershell"); !errors.Is(powershellErr, osexec.ErrNotFound) {
```

Comment:
Should we only check for poweshell when runtime.os == Windows?  

I think there's no point on checking for powershell when azd is not running in Windows

Reply:
It also doesn't hurt anybody :)